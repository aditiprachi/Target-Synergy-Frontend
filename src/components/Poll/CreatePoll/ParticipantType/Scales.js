import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Slider from '@material-ui/core/Slider';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import axios from 'axios'
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button'
import Input from '@material-ui/core/Input';
import randomColor from 'randomcolor'
const id=36

const useStyles = makeStyles({
    root: {
      width: '100%',
      height: '100%'
    },
    input: {
      width: 20,
    },
  });
  
  const Scales =() => {
    const [questions , setQuestions] = useState({questions:[]});
 
    useEffect(async () => {
      var result = await axios.get(`http://localhost:8080/QandA/${id}`)
      setQuestions({
       
        questions: result.data.questions 
      })
    },[])
    const qAndA=[];
    questions.questions.map((post,key) => (
    qAndA[key]=post.text
));
console.log(qAndA);
    const classes = useStyles();
    
    //const [inputList, setInputList] = useState([{value}]);
    const [value, setValue] = React.useState();
    const [inputList, setInputList] = useState([{ choice: "" }]);
    const handleSliderChange = (event, newValue) => {
      setValue(newValue);
    };
  
    const handleInputChange = (event,index) => {
      setValue(event.target.value === '' ? '' : Number(event.target.value));
      const { name, value } = value;
      const list = [...inputList];
      list[index][name] = value;
      setInputList(list);
    };
  
    const handleBlur = () => {
      if (value < 0) {
        setValue(0);
      } else if (value > 100) {
        setValue(100);
      }
    };
    const[data, setData]=useState({
      
        response:[]
    })
    const url="http://localhost:8080/QandAResponse"
    const submit = (e) => {

      e.preventDefault();
    //   inputList.map((resp,key)=>{
    //     data.response[key]=resp.response
    //   })
  
      const q ={
        
       responses: data.response
      }
      console.log(q)
      axios.post(url, q)
           .then(res=>{
              console.log(res.data)
            })

    }
    return (
      <div><h3 fontFamily= "Helvetica">{data5.question}</h3>
           <form onSubmit={submit} className={classes.root} noValidate autoComplete="off">
      
{qAndA.map((x, i) => {
  var colors=randomColor();
    return(
        <div>
        <Typography id="input-slider" gutterBottom>
          {qAndA[i]}
        </Typography>
        <Grid container spacing={2} alignItems="center">
          
          <Grid item xs>
            <Slider
              value={typeof value === 'number' ? value : 0}
              onChange={handleSliderChange}
              aria-labelledby="input-slider"
            
            />
          </Grid>
          <Grid item>
            <Input
              className={classes.input}
              value={qAndA[i]}
              margin="dense"
              onChange={handleInputChange}
              onBlur={handleBlur}
              inputProps={{
                step: 10,
                min: 0,
                max: 100,
                type: 'number',
                'aria-labelledby': 'input-slider',
        }}
            />

          </Grid>
        </Grid>
           
        </div>
        );
      })}
    
        <Button
        style={{ width: "248px",background:"#cc0000", color:"white" }}
        className={classes.button}
        variant="contained"
       // color="primary"
        size="large"
        fullWidth={true}
        onClick={submit}
      >Submit
      </Button>
  
       
       
       </form>
      </div>
      
    );
  }
  export default Scales