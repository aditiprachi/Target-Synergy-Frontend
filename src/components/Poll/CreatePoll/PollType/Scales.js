import React from 'react';
import PropTypes from 'prop-types';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Slider from '@material-ui/core/Slider';
import { Container } from '@material-ui/core';

const Scales =({data5,inputList}) => {
    const ScaleSlider = withStyles({
        root: {
          color: '#52af77',
          height: 8,
        },
        thumb: {
          height: 24,
          width: 24,
          backgroundColor: '#fff',
          border: '2px solid currentColor',
          marginTop: -8,
          marginLeft: -12,
          '&:focus, &:hover, &$active': {
            boxShadow: 'inherit',
          },
        },
        active: {},
        valueLabel: {
          left: 'calc(-50% + 4px)',
        },
        track: {
          height: 8,
          borderRadius: 4,
        },
        rail: {
          height: 8,
          borderRadius: 4,
        },
      })(Slider);
    return(
        <div>
            <h1>{data5.question}</h1>
        {inputList.map((x,i)=>{
            return(
            <Container key={i}>
        <b style={{textAlign: 'left'}}>{inputList[i].choice}</b>  
     <ScaleSlider  defaultValue={0}  />
</Container>)
        })}
        </div>
    );
};
export default Scales