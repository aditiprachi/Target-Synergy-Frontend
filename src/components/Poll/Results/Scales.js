import React,{useEffect,useState } from 'react'
import { Bar } from 'react-chartjs-2';
import axios from "axios";
import PropTypes from 'prop-types';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Slider from '@material-ui/core/Slider';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import randomColor from 'randomcolor'

const useStyles = makeStyles((theme) => ({
  disabled: {
    color: randomColor({luminosity: 'bright'})
  },
  root: {
    width: 300 + theme.spacing(3) * 2,
    
  },

  margin: {
    height: theme.spacing(3),
  },

}));



const Scales =(props) => {
  const classes = useStyles();
  const url =props.match.params.id;
  const [textBased , setTextBased] = useState({question:'', choices:[]})
  useEffect(async () => {
    var result = await axios.get(`https://targetsynergy-backend.herokuapp.com/SC/${url}`)
    setTextBased({
      question: result.data.question,
     choices: result.data.choices
      
    })
  },[])

  return (
    <div  style={{display: "flex", flexDirection: 'column', justifyContent: 'center', alignItems: 'center', width: '100%', height: '100%',paddingTop:"5%"}} ><h1 fontFamily= "Helvetica">{textBased.question}</h1>
           
      
{textBased.choices.map((x, i) => {
  const PrettoSlider = withStyles({
    root: {
      color: randomColor(),
      height: 8,
    },
    thumb: {
      height: 24,
      width: 24,
      backgroundColor: '#fff',
      border: '2px solid currentColor',
      marginTop: -8,
      marginLeft: -12,
      '&:focus, &:hover, &$active': {
        boxShadow: 'inherit',
      },
    },
    active: {},
    valueLabel: {
      left: 'calc(-50% + 4px)',
    },
    track: {
      height: 8,
      borderRadius: 4,
    },
    rail: {
      height: 8,
      borderRadius: 4,
    },
  })(Slider);
 
    return(
        <div style={{ justifyContent: 'center', alignItems: 'center', width: '100%' }}>
      
          <h4>{x.option}</h4>
          <div className={classes.margin} />
      
      <PrettoSlider valueLabelDisplay="on" aria-label="pretto slider" defaultValue={x.votes} />
       
          
</div>

           

        );
      })}
      </div>
    );
  }
export default Scales