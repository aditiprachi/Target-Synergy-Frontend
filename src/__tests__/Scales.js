import React from "react";
import Scales from "../components/Poll/CreatePoll/PollType/Scales";
import renderer from "react-test-renderer";

describe("Testing Snapshot for Scales.js",()=>{
    it("Renders",()=>{
        const tree = renderer.create(<Scales data5={[""]} inputList={[""]} textcolor={[""]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})