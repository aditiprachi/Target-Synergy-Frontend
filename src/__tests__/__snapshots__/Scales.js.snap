// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot for Scales.js Renders 1`] = `
<div>
  <h1
    style={
      Object {
        "color": Array [
          "",
        ],
      }
    }
  />
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <b
      style={
        Object {
          "textAlign": "left",
        }
      }
    />
    <span
      className="MuiSlider-root WithStyles(ForwardRef(Slider))-root-1 MuiSlider-colorPrimary"
      onMouseDown={[Function]}
    >
      <span
        className="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-6"
      />
      <span
        className="MuiSlider-track WithStyles(ForwardRef(Slider))-track-5"
        style={
          Object {
            "left": "0%",
            "width": "0%",
          }
        }
      />
      <input
        type="hidden"
        value="0"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-2 MuiSlider-thumbColorPrimary"
        data-index={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
          }
        }
        tabIndex={0}
      />
    </span>
  </div>
</div>
`;
